lib/logstash/inputs/redis.rb

```diff
@@ -123,9 +123,9 @@ def run(output_queue)
     # ignore and quit
   end # def run
 
-  def teardown
-    @shutdown_requested = true
-    @teardown_method.call
+  def stop
+    @stop_requested = true
```

Comment:
I was looking to that method in master! (I saw you did it in the relp input).

Will do